packages/flutter_tools/test/general.shard/flutter_validator_test.dart

```diff
@@ -180,17 +186,69 @@ void main() {
     );
 
     expect(await flutterValidator.validate(), _matchDoctorValidation(
-      validationType: ValidationType.installed,
+      validationType: ValidationType.partial,
+      statusInfo: 'Channel beta, 1.0.0, on Linux, locale en_US.UTF-8',
+      messages: containsAll(const <ValidationMessage>[
+        ValidationMessage.hint('Upstream repository https://github.com/flutter/flutter.git is not the same as FLUTTER_GIT_URL'),
+        ValidationMessage('FLUTTER_GIT_URL = https://githubmirror.com/flutter.git'),
+      ]),
+    ));
+  });
+
+  testWithoutContext('FlutterValidator fails when channel is unknown', () async {
+    final FlutterValidator flutterValidator = FlutterValidator(
+      platform: FakePlatform(localeName: 'en_US.UTF-8'),
+      flutterVersion: () => FakeFlutterVersion(
+        frameworkVersion: '1.0.0',
+        // channel is unknown by default
+      ),
+      devToolsVersion: () => '2.8.0',
+      userMessages: UserMessages(),
+      artifacts: Artifacts.test(),
+      fileSystem: MemoryFileSystem.test(),
+      processManager: FakeProcessManager.any(),
+      operatingSystemUtils: FakeOperatingSystemUtils(name: 'Linux'),
+      flutterRoot: () => 'sdk/flutter',
+    );
+
+    expect(await flutterValidator.validate(), _matchDoctorValidation(
+      validationType: ValidationType.partial,
       statusInfo: 'Channel unknown, 1.0.0, on Linux, locale en_US.UTF-8',
-      messages: contains(const ValidationMessage('FLUTTER_GIT_URL = https://githubmirror.com/flutter.git')),
+      messages: contains(const ValidationMessage.hint('Flutter version 1.0.0 on channel unknown at sdk/flutter')),
```

Comment:
This is because I want to nudge users towards hints that might cause `flutter upgrade` to fail. The tool sets channel `unknown` if the branch doesn't have an upstream.

Reply:
It doesn't seem like this message gives the user actionable information, though.

Reply:
Should we recommend the user to reinstall here(and other errors where applicable)?

Reply:
what if the if i'm tracking branch delta from my own mirror? Won't that also resolve to "Unknown"? This seems like a reasonable workflow.

Reply:
The tool parses the information from this command for the channel:
https://github.com/flutter/flutter/blob/c181e450c8dfa0080626d7572753b9c85727c0a2/packages/flutter_tools/lib/src/version.dart#L113-L114
and sets 'unknown' if the above returns nothing, which shouldn't be possible if the current branch is tracking _something_, be it a remote or a local branch.